tests/drivers/counter/counter_basic_api/boards/frdm_mcxc242.conf

```diff
@@ -0,0 +1 @@
+CONFIG_SHARED_INTERRUPTS=y
```

Comment:
I don't think we should be enabling this for this test only. Ideally, we should be selecting this Kconfig when `CONFIG_COUNTER_NXP_PIT` is enabled- but we don't need this for every platform. Probably the best way to drive this from the SOC defconfig, something like so:

```
if SOC_SERIES_MCXC # Change this to be more narrow if only MCXC242 SOC needs this enabled (IE if SOC_MCXC24)
config SHARED_INTERRUPTS
  default y if COUNTER_NXP_PIT
endif
```

Reply:
@danieldegrasse, thank you for review. SHARED_INTERRUPTS is needed only if more than 1 PIT channel is enabled.  I haven't found how to select SHARED_INTERRUPTS based on count of enabled PIT channels.
Selecting it for 1 enabled channel increases flash footprint unnecessarily (MCXC has small flash), therefor I would prefer enabling it only if necessary.

Proposal: Enable only one PIT channel in default board dts. In the test, enable 2 channels and select SHARED_INTERRUPTS in counter_basic_api/boards/frdm_mcxc242.conf, with a comment why, to demonstrate the functionality.

Would you agree on the proposal or do you prefer to select SHARED_INTERRUPTS in SOC defconfig even for 1 enabled PIT channel?

Reply:
I have placed the interrupts to the parent node as proposed by @decsny.